[quant][pt2e][be] Add QuantizationSpecBase

jerryzh168 opened the pull request:
Summary:
Make all quantization spec to inherit from the same base class in order to simplify the typing
for QuantizationAnnotation

Test Plan:
```
buck2 test mode/opt caffe2/test:quantization_pt2e -- 'caffe2/test:quantization_pt2e'
```

Reviewed By: kimishpatel

Differential Revision: D46173954

andrewor14 commented on torch/ao/quantization/_pt2e/quantizer/quantizer.py:
```diff
@@ -24,8 +29,14 @@
 
+class QuantizationSpecBase(ABC):
+    """ Base class for all types of quantization specs, this is introduced
+    to keep the typing for quantization annotation API simple
```
I feel this comment is a bit too implementation oriented. Maybe it should be something more user oriented like "Base quantization spec that allows users to specify how to quantize a tensor"

jerryzh168 replied:
OK sounds good

jerryzh168 committed:
[quant][pt2e][be] Add QuantizationSpecBase (#102582)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/102582

Make all quantization spec to inherit from the same base class in order to simplify the typing
for QuantizationAnnotation

Test Plan:
```
buck2 test mode/opt caffe2/test:quantization_pt2e -- 'caffe2/test:quantization_pt2e'
```

Reviewed By: kimishpatel, andrewor14

Differential Revision: D46173954

fbshipit-source-id: afaf3aa74834a7fc620a552c20f4febe8cdaa8ae
